tools/webtransport/h3/webtransport_h3_server.py

```diff
@@ -64,15 +162,52 @@ def _h3_event_received(self, event: H3Event) -> None:
                 self._handshake_webtransport(event, headers)
             else:
                 self._send_error_response(event.stream_id, 400)
+            self._session_stream_id = event.stream_id
 
-        if self._handler is not None:
+        if self._session_stream_id == event.stream_id and\
+           isinstance(event, WebTransportStreamDataReceived):
+            self._receive_data_on_session_stream(
+                event.data, event.stream_ended)
+        elif self._handler is not None:
             if isinstance(event, WebTransportStreamDataReceived):
-                self._handler.stream_data_received(stream_id=event.stream_id,
-                                                   data=event.data,
-                                                   stream_ended=event.stream_ended)
+                self._handler.stream_data_received(
+                    stream_id=event.stream_id,
+                    data=event.data,
+                    stream_ended=event.stream_ended)
             elif isinstance(event, DatagramReceived):
                 self._handler.datagram_received(data=event.data)
 
+    def _receive_data_on_session_stream(self, data: bytes, fin: bool) -> None:
+        self._capsule_decoder_for_session_stream.append(data)
+        if fin:
+            self._capsule_decoder_for_session_stream.final()
+        for capsule in self._capsule_decoder_for_session_stream:
+            if capsule.type == CapsuleType.DATAGRAM:
```

Comment:
I think these are known and unexpected. Do you think we should treat them as unknown?

Reply:
I don't think "unexpected" is a right word, since they're allowed by the protocol (it's just we don't need to implement them for now since Chrome doesn't send those).  How about "Unimplemented capsule type"?

Reply:
Done.